pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -110,7 +113,19 @@ def supports_derivatives(
         # backpropagation currently supported for all supported circuits
         # will later need to add logic if backprop requested with finite shots
         # do once device accepts finite shots
-        return execution_config.gradient_method == "backprop"
+        if execution_config.gradient_method == "backprop":
+            return True
+
+        if execution_config.gradient_method == "adjoint":
+            if circuit is None:
+                return True
+            try:
+                validate_and_expand_adjoint(circuit)
+                return True
+            except DeviceError:
+                return False
```

Comment:
Instead of using a try-except block here, I wonder if we should restructure `validate_and_expand_adjoint` to postpone raising the error and allow it to be used more naturally here. Not too important, and I'm sorry for not thinking of this use case earlier.

Reply:
tried something a bit different, returning the error rather than raising it and letting the calling method decide what to do with it. I don't _love_ it, but it works and avoids using try-except